html/migration-to-v2.html

```diff
@@ -5,185 +5,251 @@
+                            VERSION OF H2.</bold>
+                    </p>
+
+                    <p>
+                        Then create a fresh database <bold>USING THE NEW VERSION OF H2</bold>, then perform a SCRIPT to
+                        load your data.
+                    </p>
+
+                    <h2 id="format">MVStore file format</h2>
+
+                    <p>
+                        The MVStore file format we use (i.e. the default) is still mostly the same, but some subtle
+                        changes have been made
+                        to the undo logs,
+                        for the purposes of improving crash safety and also read/write performance.
+                    </p>
+
+                    <h2 id="datetime">Date / Time / Timestamp</h2>
+
+                    <p>
+                        TODO - Correctness.
+                    </p>
+
+                    <h2 id="general">General</h2>
+
+                    <p>
+                        There are a lot more SQL keywords now. Many SQL statements feature far better support of
+                        SQL-Standard behaviour.
+                        Some old non-standard SQL syntax support has been removed.
+                    </p>
+
+                    <p>
+                        Migrating an old database to the new version works most of the times. However, there are a
+                        couple of important changes in the new version to keep in mind:
+                    </p>
+                    <ul>
+                        <li>
+                            <b>Restricted maximum allowed length limits:</b> there are new length limits for for
+                            non-LOB data types. For example, for the VARBINARY type the maximum allowed size is now
+                            1048576 (1MB)
+                            instead of the previous 16777216 (16MB). Adjust the tables before migration.
+                        </li>
+                        <li>
+                            <b>VARCHAR length changes unit syntax:</b> The length of the VARCHAR datatype cannot
+                            be more specified using CHAR but CHARACTERS or OCTETS.
+                        </li>
+                        <li>
+                            <b>IDENTITY syntax changed when type is specified:</b> if the type for IDENTITY is
+                            specified, then the clause needs to be expanded as INTEGER GENERATED ALWAYS AS IDENTITY.
+                            Using just INTEGER IDENTITY is no more working.
+                        </li>
+                        <li>
+                            <b>Usage of statement.setTimestamp() is discouraged:</b> use setObject(parameterIndex,
+                            value) with java.time.LocalDateTime parameter instead. For example: stmt.setTimestamp(2,
+                            timestamp, this.utcCalendar); becomes stmt.setObject(2, timestamp.toLocalDateTime());.
```

Comment:
Where did you find that information?

Documentation of 1.4.200 already recommended JSR-310 data types instead, there is nothing new it that area in H2 2.0, only some bugs were fixed.

And documentation recommends usage of new data types. I doesn't recommend to convert legacy data types to modern ones for H2. If you still want to use defective by design legacy types, you can pass them to H2 as its.